docs/core/deploying/native-aot/diagnostics.md

```diff
@@ -0,0 +1,54 @@
+---
+title: AOT Diagnostics
+description: Learn about diagnostics in NativeAOT applications
+author: lakshanf
+ms.author: lakshanf
+ms.date: 08/07/2023
+---
+# Diagnostics in NativeAOT applications
+
+The long term goal for NativeAOT application diagnostics is to provide the rich diagnostic experience that developers expect out of a .NET application. .NET diagnostic experience was built over multiple releases and getting customer feedback on diagnostic was a critical part of that journey. NativeAOT wants to follow a similar path where the right diagnostic experience is built from prioritized customer feedback in multiple releases.
+
+NativeAOT applications have the following characteristics that are important to keep in mind in diagnostic scenarios:
+
+- Trimming as a first-class feature in NativeAOT application. Unused code in a NativeAOT application is stripped out of the final binary. Any unbounded reflection used by the application including its dependent libraries will be trimmed. NativeAOT compiler will generate warnings in such cases, and it is critical that these warnings are fixed. For example, suppressing any such warnings without careful analysis can lead to hard to debug production failures.
+- Symbol file issues: In NativeAOT, symbol-file-dependent diagnostics (debugging, PerfView callstacks, etc) don't work at all unless the diagnostic tool has access to the monolithic PDB that was generated when the application was compiled. In .NET, symbol-file-dependent diagnostics generally work just fine (or even work great) even if the diagnostic tool has no access to any PDBs that were generated when the application was compiled (i.e., symbols for the .NET runtime and the .NET libraries can be pulled from symbol servers, diagnostic tools can still show function names, accurate callstacks, etc even without access to the PDBs generated at application compile time).
+- NativeAOT application is neither a typical managed application (for example, no JIT) nor a typical native application (for example, has full GC support). The intent is to meet the reasonable diagnostics expectation of a product that is coming out of the .NET family and also the experience should feel familiar to someone who does production diagnostics on a C++ application.
+
+## Current NativeAOT diagnostic support
+
+The following table summarizes diagnostic features supported for native AOT deployments:
+
+| Feature | Fully Supported | Partially Supported | Not Supported |
+| - | - | - | - |
+| Build (Inner dev loop) diagnostics | <span aria-hidden="true">✔️</span><span class="visually-hidden">Fully supported</span> | | |
```

Comment:
We aren't supposed to use HTML where it can be avoided.

Reply:
I copied this from https://learn.microsoft.com/en-us/aspnet/core/fundamentals/native-aot since that seems to capture the message well. It will be great if we could use the same here.

Reply:
I'm not even sure why the HTML is needed.  But can we just link to the section it's coming from instead of trying to reproduce it? It makes it easier to maintain that way.